caddy_scraper/opensearch_document_manager.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+
+import glob
+import json
+import os
+from typing import List, Callable, Any
+
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from langchain.document_loaders import DataFrameLoader
+from langchain.embeddings import BedrockEmbeddings
+from opensearchpy import OpenSearch, helpers
+import pandas as pd
+
+
+class OpenSearchDocumentManager:
+    def __init__(self, client: OpenSearch, index_name: str = "caddy-hybrid-search-index"):
+        self.client = client
+        self.index_name = index_name
+        self.embedding_model = BedrockEmbeddings(
+            model_id="cohere.embed-english-v3", region_name="eu-west-3"
+        )
+        
+    def create_index(self):
+        index_body = {
+            "settings": {
+                "index": {"knn": True}
+            },
+            "mappings": {
+                "properties": {
+                    "text": {"type": "text"},
+                    "source": {"type": "keyword"},
+                    "domain": {"type": "keyword"},
+                    "text_vector": {
+                        "type": "knn_vector",
+                        "dimension": 1024,
+                        "method": {
+                        "engine": "faiss",
+                        "space_type": "l2",
+                        "name": "hnsw",
+                        "parameters": {}
+                    }
+                    }
+                }
+            }
+        }
+        # Delete index if it exists
+        if self.client.indices.exists(index=self.index_name):
+            self.client.indices.delete(index=self.index_name)
+        # Create new index
+        self.client.indices.create(index=self.index_name, body=index_body)
+    
+    async def async_bulk_upload(self, file_path: str, domain: str = "citizen-advice"):
+        json_files = glob.glob(os.path.join(file_path, "scrape_result_*.json"))
```

Comment:
intermediate disk step — we could also stream out of the scraper!

Reply:
That's a cool idea, we could get rid of the distinction between scraper/document_manager and just stream everything